### ISSUE TITLE
[Feature Request] Private mode

### ISSUE THREAD

**Issue by bormaxvel** [jameskokoska/Cashew#258]
Problem Statement:
When I enter the application, anyone near me can see sensitive information about my funds. 
Suppose the following scenario: I click to add a transaction through a widget on my homescreen, enter the transaction details, and confirm. Then the app automatically takes me to the main page where my account balances are displayed prominently. This poses a privacy risk as it can expose sensitive financial information to bystanders.

Proposed Solutions:
1. Auto-Close After Widget Transaction: Modify the workflow so that upon completing a transaction via the widget (Quick Add feature), the app automatically closes instead of redirecting to the main page. This prevents accidental exposure of financial information.
2. Private Mode Option: Introduce a private mode where all financial amounts are obscured by default. Users can reveal this information by tapping on the obscured fields. Ideally, this mode would be global — once a private field is tapped, it would reveal all private fields during that session. If implementing this feature across all data points is complex, a simpler alternative could be to obscure only the account balances. This functionality is similar to the spoiler feature in the Telegram messenger, providing a convenient layer of privacy.

Additional Context:
This feature would align with privacy enhancements seen in other applications, such as banking apps, but cannot be demonstrated here due to privacy reasons. However, I can provide a demonstration using the Telegram app's spoiler function to illustrate how this might work in practice.
https://github.com/jameskokoska/Cashew/assets/90153047/b2ab1457-609d-4b6c-8121-2565375f0d30

**Comment by jameskokoska** [comment]
Hi there, thank you for the feature request. It's hard to see how the proposed solutions would fit into the application and user workflow. Currently you can modify the home page sections you want displayed.

Because most homepage widgets have a tap action already, from a use case perspective it's quite unintuitive to have the initial tap reveal the hidden amount, with the following tap processing the tap action.

The overarching suggestion and solution is to make sure you use the application in a private place if you want to keep sensitive information hidden. 

Edit: The main usability issue is - amounts are hidden, how does the user easily
unhide the amounts at any point and how would this be communicated to the user without always being in the way.

**Comment by bormaxvel** [comment]
@jameskokoska 
> homepage widgets have a tap action already

Indeed, I hadn't thought about this in relation to the tap function for viewing hidden information. It probably won't really work. 

But what about option 1? It wouldn't make the user experience worse, and it would completely solve the privacy problem when adding a transaction. Because the idea of the application is that we need to add data much more often than analyze it, so is it possible to add to the widget settings the choice of action after adding a transaction? You just need to somehow add tracking to the routing logic that we went to the addTransactionPage from the home screen widget, and if so, the application could simply be closed. In this case, the UX would be improved, because the program would return us to the previous state from which we entered this screen - the home screen.
